### ISSUE TITLE
Please add thumbnail support which GTK sadly lacks

### ISSUE THREAD

**Issue by ghost** [thesquash/stlwrt#5]
It's a well known issue that GTK has no thumbnail support in the file picker dialogue. However, there is a positive solution, someone already made a patch along with a forked GTK2/3 that has thumbnail support in the file picker. All you need to do is two things:
1. Apply the patch provided here: https://github.com/Dudemanguy/gtk/tree/gtk2-filechooser-icon-view
2. Somehow have the file picker generate the thumbnails when a directory is loaded as noted at the bottom of the link because you would otherwise need to patch glib2 as well.

**Comment by thesquash** [comment]
I completely agree over this change.  I just downloaded that patch; I am now looking over the patch and STLWRT-izing it, and additionally I *might possibly* change the look and feel of the new feature slightly, just to make it fit in with the file chooser a little better.
